src/test/unit/algorithms/SpatialPoolerTest.cpp

```diff
@@ -1108,145 +1108,6 @@ TEST(SpatialPoolerTest, testCalculateOverlapPct) {
   }
 }
 
-TEST(SpatialPoolerTest, testIsWinner) {
-  UInt numInputs = 10;
-  UInt numColumns = 5;
-  SpatialPooler sp({numInputs}, {numColumns});
-
-  vector<pair<UInt, Real>> winners;
-
-  UInt numWinners = 3;
-  Real score = -5;
-  ASSERT_FALSE(sp.isWinner_(score, winners, numWinners));
-  score = 0;
-  ASSERT_TRUE(sp.isWinner_(score, winners, numWinners));
-
-  pair<UInt, Real> sc1;
-  sc1.first = 1;
-  sc1.second = 32;
-  pair<UInt, Real> sc2;
-  sc2.first = 2;
-  sc2.second = 27;
-  pair<UInt, Real> sc3;
-  sc3.first = 17;
-  sc3.second = 19.5;
-  winners.push_back(sc1);
-  winners.push_back(sc2);
-  winners.push_back(sc3);
-
-  numWinners = 3;
-  score = -5;
-  ASSERT_TRUE(!sp.isWinner_(score, winners, numWinners));
-  score = 18;
-  ASSERT_TRUE(!sp.isWinner_(score, winners, numWinners));
-  score = 18;
-  numWinners = 4;
-  ASSERT_TRUE(sp.isWinner_(score, winners, numWinners));
-  numWinners = 3;
-  score = 20;
-  ASSERT_TRUE(sp.isWinner_(score, winners, numWinners));
-  score = 30;
-  ASSERT_TRUE(sp.isWinner_(score, winners, numWinners));
-  score = 40;
-  ASSERT_TRUE(sp.isWinner_(score, winners, numWinners));
-  score = 40;
-  numWinners = 6;
-  ASSERT_TRUE(sp.isWinner_(score, winners, numWinners));
-
-  pair<UInt, Real> sc4;
-  sc4.first = 34;
-  sc4.second = 17.1;
-  pair<UInt, Real> sc5;
-  sc5.first = 51;
-  sc5.second = 1.2;
-  pair<UInt, Real> sc6;
-  sc6.first = 19;
-  sc6.second = 0.3;
-  winners.push_back(sc4);
-  winners.push_back(sc5);
-  winners.push_back(sc6);
-
-  score = 40;
-  numWinners = 6;
-  ASSERT_TRUE(sp.isWinner_(score, winners, numWinners));
-  score = 12;
-  numWinners = 6;
-  ASSERT_TRUE(sp.isWinner_(score, winners, numWinners));
-  score = 0.1;
-  numWinners = 6;
-  ASSERT_TRUE(!sp.isWinner_(score, winners, numWinners));
-  score = 0.1;
-  numWinners = 7;
-  ASSERT_TRUE(sp.isWinner_(score, winners, numWinners));
-}
-
-TEST(SpatialPoolerTest, testAddToWinners) {
```

Comment:
ok to remove the code that tests the specific (private) methods. But you should introduce a test that validates the winners functionality

Reply:
Ok, I can fix these things